src/screens/DiscoveryScreenV2/index.js

```diff
@@ -116,8 +147,18 @@ const DiscoveryScreenV2 = ({route}) => {
     };
 
     const fetchDiscoveryInitialTopics = async () => {
-      const initialData = await DiscoveryRepo.fetchInitialDiscoveryTopics();
-      DiscoveryAction.setDiscoveryInitialTopics(initialData.suggestedTopics, discoveryDispatch);
+      const initialData = await DiscoveryRepo.fetchInitialDiscoveryTopics(
+        50,
+        parseInt(topicPage.currentPage, 10)
+      );
+      setTopicPage({
+        currentPage: initialData.page,
+        totalPage: initialData.total_page
+      });
+      DiscoveryAction.setDiscoveryInitialTopics(
+        [...discoveryData.initialTopics, ...initialData.suggestedTopics],
+        discoveryDispatch
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Again, this code block is repeating the same logic as in `fetchDiscoveryInitialUsers` and `fetchDiscoveryInitialDomains`. Consider using the generic function suggested above.
```diff
- const fetchDiscoveryInitialTopics = async () => {
-   const initialData = await DiscoveryRepo.fetchInitialDiscoveryTopics(
-     50,
-     parseInt(topicPage.currentPage, 10)
-   );
-   setTopicPage({
-     currentPage: initialData.page,
-     totalPage: initialData.total_page
-   });
-   DiscoveryAction.setDiscoveryInitialTopics(
-     [...discoveryData.initialTopics, ...initialData.suggestedTopics],
-     discoveryDispatch
-   );
- };

+ // Use the generic function here
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->